src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateHoldingEventHandler.java

```diff
@@ -201,9 +207,28 @@ private void prepareDataAndReInvokeCurrentHandler(DataImportEventPayload dataImp
     });
   }
 
-  private void constructDataImportEventPayload(CompletableFuture<DataImportEventPayload> future, DataImportEventPayload dataImportEventPayload, HoldingsRecord holding) {
-    dataImportEventPayload.getContext().put(HOLDINGS.value(), Json.encodePrettily(holding));
-    future.complete(dataImportEventPayload);
-    future.complete(dataImportEventPayload);
+  private void constructDataImportEventPayload(CompletableFuture<DataImportEventPayload> future, DataImportEventPayload dataImportEventPayload, ItemCollection itemCollection, HoldingsRecord holding) {
+    HashMap<String, String> payloadContext = dataImportEventPayload.getContext();
+    payloadContext.put(HOLDINGS.value(), Json.encodePrettily(holding));
+    if(payloadContext.containsKey(ITEM.value())) {
+      updateDataImportEventPayloadItem(future, dataImportEventPayload, itemCollection);
+    } else {
+      future.complete(dataImportEventPayload);
+    }
+  }
+
+  private void updateDataImportEventPayloadItem(CompletableFuture<DataImportEventPayload> future, DataImportEventPayload dataImportEventPayload, ItemCollection itemCollection) {
```

Comment:
Can you add code coverage to this method?

Reply:
@julianladisch I'm on vacation. But if necessary, I can make time.

Reply:
ready

Reply:
Thanks you!